extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeSearchQueryHandlerFactory.java

```diff
@@ -2,19 +2,83 @@
 
 import org.schabi.newpipe.extractor.exceptions.ParsingException;
 import org.schabi.newpipe.extractor.linkhandler.SearchQueryHandlerFactory;
+import org.schabi.newpipe.extractor.utils.Base64;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import java.io.UnsupportedEncodingException;
 import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
 public class YoutubeSearchQueryHandlerFactory extends SearchQueryHandlerFactory {
 
     public static final String CHARSET_UTF_8 = "UTF-8";
 
-    public static final String VIDEOS = "videos";
-    public static final String CHANNELS = "channels";
-    public static final String PLAYLISTS = "playlists";
-    public static final String ALL = "all";
+    public enum FilterType {
+        Content((byte)0x10),
+        Time((byte)0x08),
+        Duration((byte)0x18);
+
+        private final byte value;
+
+        FilterType(byte value) {
+            this.value = value;
+        }
+    }
+
+    public enum Filter {
+        All(FilterType.Content,(byte)0),
+        Video(FilterType.Content,(byte)0x01),
+        Channel(FilterType.Content,(byte)0x02),
+        Playlist(FilterType.Content,(byte)0x03),
+        Movie(FilterType.Content,(byte)0x04),
+        Show(FilterType.Content,(byte)0x05),
+
+        Hour(FilterType.Time,(byte)0x01),
```

Comment:
I am not sure what you mean but this works out if the box. Albeit it looks kinda bad:

![Screenshot_1553365615](https://user-images.githubusercontent.com/5329780/54870295-a6484f00-4dca-11e9-86e8-65d8674fbf34.png)

Reply:
Yes this is the issue i ment, it is not possible to add or remove filters like key values. This messes up the UI as you see. The UI should be made ready to handle these filters.

Reply:
Maybe we could implement the filters at 2 levels? We would initially ask the user to select a `FilterType` and after they have chosen that, we can ask them the exact filter they wish to apply falling under that FilterType? 

So, clicking on the menu icon the first time, will load a list containing `Content, Time, Duration, Feature`. If the user chooses Duration, then we will show a second list containing `Short, Long`. And when the user chooses one of those options, then the search is triggered. Does that make sense?

Reply:
Let's finish this PR.  🎉 Thanks for being patient!

I am working on a proper design for the filters. This might take until next Sunday, because the new semesters is about to start and thus there are a bunch of other things to do. But I would not recommend to have endless lists. They come with bad UI/UX, because it is hard to get all the (important) information.

FYI: 
- [6 Best Practices forMobile App Search Filtering](https://www.raywenderlich.com/718-6-best-practices-for-mobile-app-search-filtering)
- [Filter and Sorting Best Practices on Mobile](https://medium.com/@thierrymeier/filtering-and-sorting-best-practices-on-mobile-61626449cec)
- [Search, Sort and Filter - Mobile Design Pattern Gallery](https://www.oreilly.com/library/view/mobile-design-pattern/9781449368586/ch04.html)
- [Collection of various screenshots](https://www.pinterest.de/darylhumphrey9/filter-mobile-ux/)
- [animations](https://uimovement.com/tag/filter/)

Reply:
@TobiGr Of course, take your time. I was just surprised and happy that it worked functionally out of the box.

Reply:
I put my UI ideas into [this issue](https://github.com/TeamNewPipe/NewPipe/issues/2251) in the app repo. I guess a UI discussion fits better into the other repo.